e2e-test/src/test/java/io/cdap/plugin/distinct/stepsdesign/DistinctAnalytics.java

```diff
@@ -66,4 +69,27 @@ public void verifyDistinctPluginInLineErrorMessageForIncorrectFields(String fiel
   public void enterDistinctPluginOutputSchema(String jsonOutputSchema) {
     DistinctActions.enterOutputSchema(jsonOutputSchema);
   }
+
+  @Then("Validate target table {string} records count is equal to source table {string} records count with distinct " +
+    "fields {string}")
+  public void validateTargetTableRecordsCountIsEqualToSourceTableRecordsCountWithDistinctFields(String targetTable,
+                                                                                 String sourceTable, String fields)
+    throws IOException, InterruptedException {
+      Optional<String> result = BigQueryClient
+        .getSoleQueryResult("SELECT COUNT(*) FROM (SELECT  distinct "  +  PluginPropertyUtils.pluginProp(fields) +
+                              " as records FROM `"
+      + (PluginPropertyUtils.pluginProp("projectId")) + "."
+      + (PluginPropertyUtils.pluginProp("dataset")) + "."
+      + PluginPropertyUtils.pluginProp(sourceTable) + "`)");
+    int distinctCountBQTable = 0;
```

Comment:
line 84 to 89 can be replaced with single line ->

`int distinctCountBQTable = result.map(Integer::parseInt).orElse(0);`

Reply:
Change done